src/asmcnc/comms/router_machine.py

```diff
@@ -1053,18 +1053,144 @@ def disable_theateam(self):
+                compensated_RPM = 0
+            elif compensated_RPM > 25000:
+                compensated_RPM = 25000
+
+            return compensated_RPM
+        
+        else:
+            log("Requested RPM {} outside of range for 120V spindle (10000 - 25000)".format(target_rpm))
+            return 0
+
+    def correct_rpm_for_230(self, target_rpm):
+        """
+        Compensates for the desparity in set and actual spindle RPM for a 230V spindle.
+
+        Args:
+            target_rpm (int): The target RPM to be corrected.
+
+        Returns:
+            int: The corrected RPM value.
+        """
+        # For conversion maths see https://docs.google.com/spreadsheets/d/1Dbn6JmNCWaCNxpXMXxxNB2IKvNlhND6zz_qQlq60dQY/edit#gid=1507195715
+
+        if 4000 <= target_rpm <= 25000:
+        
+            compensated_RPM = int(round((target_rpm - 1886) / 0.95915))
+
+            if compensated_RPM < 0:
+                log("Calculated RPM {} too low for 230V spindle, setting to 0".format(target_rpm))
+                compensated_RPM = 0
+            elif compensated_RPM > 25000:
+                compensated_RPM = 25000
+
+            return compensated_RPM
+    
+        else:
+            log("Requested RPM {} outside of range for 230V spindle (4000 - 25000)".format(target_rpm))
+            return 0
+        
+    def correct_rpm(self, requested_rpm, spindle_voltage = None):
+        """
+        Compensates for the desparity in set and actual spindle RPM for a spindle.
+
+        For use outside of router_machine.py
+
+        Args:
+            requested_rpm (float): The RPM value to be corrected.
+            voltage (int, optional): The spindle voltage. Defaults to spindle_voltage.
+
+        Returns:
+            float: The corrected RPM value.
+
+        Raises:
+            ValueError: If the spindle voltage is not recognised.
+        """
+        
+        if spindle_voltage is None:
+            spindle_voltage = self.spindle_voltage # Use spindle voltage set by user in maintenance app
+
+        if spindle_voltage in [110, 120]:            
+            rpm_to_set = self.correct_rpm_for_120(requested_rpm)                
+        
+        elif spindle_voltage in [230, 240]:
+            rpm_to_set = self.correct_rpm_for_230(requested_rpm)
+        
+        else:
+            raise ValueError('Spindle voltage: {} not recognised'.format(spindle_voltage))
+        
+        log("Requested RPM: "+ str(requested_rpm) + " Compensated RPM: " + str(rpm_to_set) + " Voltage: " + str(spindle_voltage))
+
+        return rpm_to_set
+        
+    def turn_on_spindle(self, rpm=None):
+        """
+        This method sends the command 'M3' to the Z Head to turn on the spindle at a given speed.
+
+        No RPM compensation occurs in this command as this is captured and handled by compensate_spindle_speed_command() in the SerialConnection object 
```

Comment:
Could this be added to backlog? I don’t like how the speed is changed from a direct call to s.write_command.

Reply:
(From a dev perspective I wouldn’t expect my call to write command to be altered in any way - just sent)

Reply:
Understood, just something to have in the backlog so we can think about it in the future